Documentation/MapStyleEditor.md

```diff
@@ -0,0 +1,88 @@
+
+If you enable the option to update while editing, any RegionMap instances in the current scene that use this MapStyle will be rebuilt using the current MapStyle settings whenever a change is made. **IMPORTANT:** These automatic updates will overwrite any manual edits you have made to the map objects in the scene.
+
+Layer List
+----------
+
+![MapStyle layer list editor](Images/MapStyleLayerList.png)
+
+Next is a list of "Layers". Each Layer in a Map Style represents a grouping of objects in the map data (these objects are called features) and the settings that will be applied to those objects when they are downloaded.
+
+You can add a new layer with the 'Add Layer' button and remove one or more layers by selecting them in the list and pressing the 'Remove Selected' button. The ordering of Layers in the list does not affect the construction of the map.
+
+Layers can be renamed in the list by selecting one and pressing Enter. It's often helpful to name Layers according to the grouping of features that you intend to include in the Layer. For example, if you wanted a Layer to style buildings of a certain height you could name the layer "Tall Buildings".
+
+Layer Properties
+----------------
+
+![MapStyle layer properties editor](Images/MapStyleLayerProperties.png)
+
+When a layer is selected its properties will be displayed below the list.
+
+### Feature Collections
+
+Features in the map data are organized into broad groupings called "feature collections". Use this drop-down to select which feature collection(s) this Layer will apply to.
+
+### Matchers
+
+To narrow down the set of features that this Layer will apply to, you can add one or more Matchers in the form of a list.
+
+Each item in the list is a separate Matcher, which can be one of several types selectable from the drop-down.
+
+Type | Description
+-----|------------
+None | Matches all features.
+Property | Matches features that contain the property named in 'Property Key', regardless of the value of that property.
+Property Range | Matches features have the property named in 'Property Key' with a value >= and < the two range parameters.
+Property Regex | Matches features have the property named in 'Property Key' with a value that matches the regular expression in the 'With Pattern' parameter.
+Property Value | Matches features have the property named in 'Property Key' with a value equal to the 'With Value' parameter.
+
+### Combine Matchers
+
+If you have more than one Matcher in the Matchers list, they can be logically combined with one of several options.
+
+Option | The Layer will apply to features that ...
+-------|------------------------------------------
+All Of | match all of the Matchers in the list
+Any Of | match any of the Matchers in the list
+None Of | match any of the Matchers in the list
+
+### Style
+
+Each of the features in this Layer will be turned into renderable meshes using the builders enabled in the Style section.
+
+### Polygon Builder
```

Comment:
Probably make sense to mention the default values here.

Reply:
Example in `extrusion` or `UV Mode` properties.

Reply:
Actually I think it's simplest to not name the default values. Unlike in a Tangram scene file, values can't be omitted. So the current value of a field is always present and visible, whether it is default or not.